src/runs.rs

```
@@ -0,0 +1,341 @@
+            Some(process) => DaemonRunStatus::from_process_status(process.status()),
+            // if no process is associated with the pid, it means the process is terminated
+            None => DaemonRunStatus::Terminated,
+        }
+    }
+
+    pub fn pid_file_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.pid", self.name))
+    }
+
+    pub fn stdout_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.out", self.name))
+    }
+
+    pub fn stderr_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.err", self.name))
+    }
+
+    pub fn infos_file_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.infos.json", self.name))
+    }
+
+    pub fn read_pid(&self) -> Option<Pid> {
+        if !self.pid_file_path().exists() {
+            return None;
+        }
+
+        let pid = std::fs::read_to_string(self.pid_file_path())
+            .expect("Failed to read pid file")
+            .trim()
+            .parse::<Pid>()
+            .expect("Failed to parse pid file content as u32");
+
+        Some(pid)
+    }
+
+    pub fn read_infos(&self) -> Option<DaemonRunInfos> {
+        if !self.infos_file_path().exists() {
+            return None;
+        }
+
+        let infos_json = std::fs::read_to_string(self.infos_file_path()).unwrap();
+        let infos: DaemonRunInfos = serde_json::from_str(&infos_json).unwrap();
+
+        Some(infos)
+    }
+
+    pub fn read_stdout(&self) -> String {
+        std::fs::read_to_string(self.stdout_path()).unwrap()
+    }
+
+    pub fn read_stderr(&self) -> String {
+        std::fs::read_to_string(self.stderr_path()).unwrap()
+    }
+
+    pub fn start(&self, task: Vec<String>) -> miette::Result<()> {
+        #[cfg(unix)]
+        {
+            // Create stdout and stderr files
+            let stdout = std::fs::File::create(self.stdout_path()).unwrap();
+            let stderr = std::fs::File::create(self.stderr_path()).unwrap();
```

I think these `unwrap`s should return an error instead. If for instance the disk is full this might fail. We should gracefully handle that.

---

I replaced `unwrap()` by `.into_diagnostic()?` - is that the correct way to do that in that context?